StudioTBA/CoronaIO
# Dynamic generation of police and humans in Game

## What issue is this PR solving?
Closes: #51 

## Description of change
- Spawn points have been set on the tiles.
- When the Game scene starts, the `Game Manager` will create the tiles (which come with the spawns) and then it will create the civilians.
- Different probabilities have been set for spawning, and what kind of civilian should spawn. These are modifiable via the inspector in `Game Manager`

NOTES:
1. I did not add spawn points to obstruction tiles since `@Alex-Hayeur` will be altering them in #73. Nevertheless, adding them is trivial. It is a matter of adding the `Spawn` prefab to the desired tile prefab. After that, upon start, the `CivilianGenerator` script will pull the new spawns.
2. The human and police's scale is enormous for Game scene. I left them as they were (50, 50, 50); however, for the scene to properly render, set them to (1,1,1).

## How to reproduce the changes
1. Go to Human prefab, set scale to (1,1,1) and position to (0,1,0).
2. Do the same for Police prefab
3. Run Game
4. See the civilians that have been created. Each time their positions should be different.

## Screenshots or GIFs
### Floor scale 30
<img width="796" alt="Screen Shot 2020-04-11 at 4 55 37 PM" src="https://user-images.githubusercontent.com/10281272/79054830-2465d680-7c16-11ea-8d69-fb7c2ddfb3bc.png">

### Floor scale 70
<img width="799" alt="Screen Shot 2020-04-11 at 4 56 29 PM" src="https://user-images.githubusercontent.com/10281272/79054834-3182c580-7c16-11ea-835c-187ccdf6a2d9.png">
